scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/job/factories/spi/stax/StaxJobValidatorServiceProvider.java

```diff
@@ -61,18 +59,15 @@ public StaxJobValidatorServiceProvider() {
     }
 
     @Override
-    public File validateJob(File jobFile) throws JobValidationException {
-        String findSchemaByNamespaceUsed;
+    public void validateJob(InputStream jobInputStream) throws JobValidationException {
         try {
-            findSchemaByNamespaceUsed = findSchemaByNamespaceUsed(jobFile);
+            byte[] bytes = ValidationUtil.getInputStreamBytes(jobInputStream);
+            String findSchemaByNamespaceUsed = findSchemaByNamespaceUsed(new ByteArrayInputStream(bytes));
             InputStream schemaStream = this.getClass().getResourceAsStream(findSchemaByNamespaceUsed);
-            ValidationUtil.validate(jobFile, schemaStream);
+            ValidationUtil.validate(new ByteArrayInputStream(bytes), schemaStream);
```

Comment:
possible un closed ByteArrayInputStream

Reply:
Yes thank you, I added a try-with-resource at the four places where a ByteArrayInputStream is created.